src/renderer/containers/Account/DeleteAccountModal/index.tsx

```diff
@@ -1,24 +1,36 @@
 import React, {FC} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {useHistory} from 'react-router-dom';
 
 import Icon, {IconType} from '@renderer/components/Icon';
 import Modal from '@renderer/components/Modal';
-import {useBooleanState} from '@renderer/hooks';
+import {getActiveBankConfig, getManagedAccounts} from '@renderer/selectors';
+import {unsetManagedAccount} from '@renderer/store/app';
+import {AppDispatch} from '@renderer/types';
+import {formatPathFromNode} from '@renderer/utils/address';
 
 import './DeleteAccountModal.scss';
 
 interface ComponentProps {
+  accountNumber: string;
   toggleDeleteModal(): void;
 }
 
-const DeleteAccountModal: FC<ComponentProps> = ({toggleDeleteModal}) => {
-  const [submittingDeleteModal, , setSubmittingDeleteModalTrue, setSubmittingDeleteModalFalse] = useBooleanState(false);
+const DeleteAccountModal: FC<ComponentProps> = ({accountNumber, toggleDeleteModal}) => {
+  const activeBankConfig = useSelector(getActiveBankConfig);
+  const dispatch = useDispatch<AppDispatch>();
+  const history = useHistory();
+  const managedAccounts = useSelector(getManagedAccounts);
 
   const handleSubmit = async (): Promise<void> => {
-    setSubmittingDeleteModalTrue();
-    setTimeout(() => {
-      setSubmittingDeleteModalFalse();
+    const account = managedAccounts[accountNumber];
+    dispatch(unsetManagedAccount(account));
+
+    if (activeBankConfig) {
+      history.push(`/banks/${formatPathFromNode(activeBankConfig)}/overview`);
+    } else {
       toggleDeleteModal();
-    }, 1000);
+    }
```

Comment:
hmm i get why you did this, but should there ever be a case when you are here on this page, and `activeBankConfig` is null?